azerothcore/azerothcore-wotlk
# Fix(script/LBRS)  urok doomhowl improvements

"Urok enforcer" and "Urok ogre magus" (entry 10601, 10602) appear neutral.
They should be hostile. Fixed.
Urok has wrong yell on aggro "me smash, you die". Should be fixed to his yell.
Urok was not casting intimidating shout. Added. Timer is TBD

Partial fix for AzerothCore 7364



## Changes Proposed:
- Add intimidating shout to the casts on timer.
- changed Urok enforcer and Urok ogre mage faction (making them hostile at the same time). Removed civilian flag as well.
- move the "say" to initialize. Added an on-spawn aggro of nearest target.


## Issues Addressed:

- Partial fix of https://github.com/azerothcore/azerothcore-wotlk/issues/7364
- partial fix for https://github.com/azerothcore/azerothcore-wotlk/issues/8712 , not sure the despawn/resummon part is a bug. I can resummon him, but you should only ever get one head in the dungeon...

## SOURCE:

https://youtu.be/AY-m_3lGiqg?t=97 . It's in the issue too.

## Tests Performed:

- Builds. 
- Fear is casted. 
- Urok yells on spawn.
- Other mobs appear red
- Urok attacks on spawn


## How to Test the Changes:


1. Build and apply changes
2. go to lbrs
.additem 12533
.additem 12534
.go o 99791
Fight through event
3. verify changes

1. start the event but stay in gm mode (or stealthed, or anything so the mobs don't aggro).
2. The boss should eventually despawn, and do so before the mobs despawn.

## Known Issues and TODO List:


- [ ] mobs attacking the pike
- [ ] Is the timer ok? Do we have any source on the time it should be?


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
